etc/copy-packetline.sh

```diff
@@ -0,0 +1,153 @@
+    fail 'git-rev-parse failed to find top-level dir'
+  root="${root_padded%$'\n.'}"
+
+  cd -- "$root"
+}
+
+function merging () {
+  local git_dir_padded git_dir
+
+  # Find the .git directory. (Padding is for the trailing-newline case.)
+  git_dir_padded="$(git rev-parse --git-dir && echo -n .)" ||
+    fail 'git-rev-parse failed to find git dir'
+  git_dir="${git_dir_padded%$'\n.'}"
+
+  test -e "$git_dir/MERGE_HEAD"
+}
+
+function target_dir_status () {
+  git status --porcelain --ignored=traditional -- "$target_dir" ||
+    fail 'git-status failed'
+}
+
+function check_target_dir () {
+  if ! test -e "$target_dir"; then
+    # The target does not exist on disk, so nothing will be lost. Proceed.
+    return
+  fi
+
+  if merging; then
+    # In a merge, it would be confusing to replace anything at the target.
+    if target_dir_status | grep -q '^'; then
+      fail 'target exists, and a merge is in progress'
+    fi
+  else
+    # We can lose data if anything of value at the target is not in the index.
+    # (This includes unstaged deletions, for two reasons. One is that we could
+    # lose track of which files had been deleted. More importantly, replacing a
+    # staged symlink or regular file with an unstaged directory is shown by
+    # git-status as only a deletion, even if the directory is non-empty.)
+    if target_dir_status | grep -q '^.[^ ]'; then
+      fail 'target exists, with unstaged changes or ignored files'
+    fi
+  fi
+}
+
+function first_line_ends_crlf () {
+  # This is tricky to check portably. In Cygwin-like environments including
+  # MSYS2 and Git Bash, most text processing tools, including awk, sed, and
+  # grep, automatically ignore \r before \n. Some ignore \r everywhere. Some
+  # can be told to keep \r, but in non-portable ways that may affect other
+  # implementations. Bash ignores \r in some places even without "-o igncr",
+  # and ignores \r even more with it, including in all text from command
+  # substitution. Simple checks may be non-portable to other OSes. Fortunately,
+  # tools that treat input as binary data are exempt (even cat, but "-v" is
+  # non-portable, and unreliable in general because lines can end in "^M").
+  # This may be doable without od, by using tr more heavily, but it could be
+  # hard to avoid false positives with unexpected characters or \r without \n.
+
+  head -n 1 -- "$1" |  # Get the longest prefix with no non-trailing \n byte.
+    od -An -ta |       # Represent all bytes symbolically, without addresses.
+    tr -sd '\n' ' ' |  # Scrunch into one line, so "cr nl" appears as such.
+    grep -q 'cr nl$'   # Check if the result signifies a \r\n line ending.
+}
+
+function make_header () {
+  local source_file endline
+
+  source_file="$1"
+  endline="$2"
+
+  # shellcheck disable=SC2016  # The backticks are intentionally literal.
+  printf '//! DO NOT EDIT - this is a copy of %s. Run `just copy-packetline` to update it.%s%s' \
```

Comment:
I just realized that this crate gets published, and we'd not want actual docstrings for this, but just a comment.

Reply:
Is it okay for a normal `//` comment to precede a top-level `//!` documentation comment? (There is one file in that subtree, `lib.rs`, where that would happen.) If so, then this is very easy to change; if not, it should still be feasible to fix without too much trouble.

Reply:
I think it's OK, and validated it.

Reply:
Great--I'll make that change shortly.

Reply:
I have changed the `//!` comments to `//` (modifying and rerunning the script).